txtorcon/torcontrolprotocol.py

```diff
@@ -103,6 +104,25 @@ def unlisten(self, cb):
 
     def got_update(self, data):
         # print self.name,"got_update:",data
+        if self.use_stem:
+            import stem.response
+            response_data = ""
+            # Add appropriate headers for parsing into stem.response.ControlMessage instance
+            if self.name == "CIRC":
+                response_data = "650 " + self.name + " " + data.replace('\n', ' ') + "\r\n"
+            elif self.name == "NEWCONSENSUS" or self.name == "NS":
+                lines = data.split('\n')
```

Comment:
The `NEWCONSENSUS/NS` events need a `650+` in the headers and also every line in the data should be separated by a `\n`. In the end of all the lines, a `\r\n` is required. However for CIRC events, all the lines of the data should not be separated by any `\n`. For the rest of the events, we simply need to add the 650 header alongwith the event name and a `\r\n` in the end.
If the events are not built in such a manner, it will fail to get parsed into the Stem class.